homeassistant/components/stt/__init__.py

```diff
@@ -36,33 +49,171 @@
 
 __all__ = [
     "async_get_provider",
+    "async_get_provider_entity",
     "AudioBitRates",
     "AudioChannels",
     "AudioCodecs",
     "AudioFormats",
     "AudioSampleRates",
     "DOMAIN",
     "Provider",
+    "ProviderEntity",
     "SpeechMetadata",
     "SpeechResult",
     "SpeechResultState",
 ]
 
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def async_get_provider_entity(
+    hass: HomeAssistant, entity_id: str
+) -> ProviderEntity | None:
+    """Return provider entity."""
+    component: EntityComponent[ProviderEntity] = hass.data[DOMAIN]
+
+    return component.get_entity(entity_id)
+
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up STT."""
+    component = hass.data[DOMAIN] = EntityComponent[ProviderEntity](
+        _LOGGER, DOMAIN, hass
+    )
+
+    # Cancel the retry setup timer on shutdown (temporary workaround)
```

Comment:
We're discussing how to solve this for integrations that don't support `async_setup_platform` and don't want to call `EntityComponent.async_setup`.

Reply:
This can be adjusted now.

Reply:
We can now use the new method added here:
https://github.com/home-assistant/core/pull/90938